src/app/data-planes/sources.ts

```diff
@@ -73,6 +81,30 @@ export const sources = (api: KumaApi) => {
       return api.getDataplaneData({ mesh, dppName: name, dataPath })
     },
 
+    '/meshes/:mesh/dataplanes/:name/gateway': (params: DetailParams, source: Closeable) => {
+      source.close()
+
+      const { mesh, name } = params
+
+      return api.getMeshGatewayDataplane({ mesh, name })
+    },
+
+    '/meshes/:mesh/dataplanes/:name/sidecar-dataplanes': (params: DetailParams, source: Closeable) => {
+      source.close()
+
+      const { mesh, name } = params
+
+      return api.getSidecarDataplanePolicies({ mesh, name })
```

Comment:
Gotcha, should the URI for this source mention `/policies` then?

Reply:
Yeah, I think so. It’s a little odd with `getMeshGatewayDataplane` which uses the same API endpoint but returns a `MeshGatewayDataplane` object. But nothing to be done about it. I’ll rename the sidecar DP policies source.

Reply:
Changed this now. Also noticed I had accidentally used the `.../sidecar-dataplane-policies` source for the gateway ones, too and fixed that.

Reply:
Hmm ok, umm I have a lot of thoughts here as I'm realizing what is happening here. Not sure whether we should sort this out here or circle back 🤔 

Umm some questions to help me:

1. Is `getMeshGatewayDataplane` also for getting policies? (I took a quick look at the actual HTTP API URL and it ends in `/policies` so I'm guessing it is for that)
2. We quite heavily distinguish between `proxies/dataplanes/sidecars` (specifically `sidecars`) and `gateways` in the application, not only from a user perspective but also pretty heavily in the codebase (we have a specific `gateways` module). I'm wondering if the source for getting policies for a gateway should be in the `gateways` module. wdyt?

Ok I just saw your last change and I do think we need to change something here now.

Ideally (if I've understood this correctly), the source URIs should be something like:

```
'/mesh/:mesh/gateway/:name/policies'
'/mesh/:mesh/dataplane/:name/policies'
```

I know dataplane isn't specifically saying `sidecar`, but at least we are specifically saying we have two separate things here and they more or less match the name of our modules.

Note: I've used the singular for above as that is my ideal, not too concerned right now to keep using plurals for retrieving single things so I suppose this would be fine for now:

```
'/meshes/:mesh/gateways/:name/policies'
'/meshes/:mesh/dataplanes/:name/policies'
```

Anyway let me know on those questions, I may have misunderstood what these are for and answers to those would help.

Reply:
```
'/meshes/:mesh/policies/for/gateway/:name'
'/meshes/:mesh/policies/for/dataplane/:name'
```

^ that is another alternative and means that these sources would be part of the `policies` module. I've been doing this form for a while now and I think I like it better.

Anyway lemme know on the above

Reply:
> 1\. Is `getMeshGatewayDataplane` also for getting policies? (I took a quick look at the actual HTTP API URL and it ends in `/policies` so I'm guessing it is for that)

Partially, yes, but the object returned isn’t _just_ policies. It’s a `MeshGatewayDataplane` with `gateway`, `listeners`, and `policies` properties. I’m still fine with changing the data source paths, but it’s definitely an odd API landscape we’re dealing with here.

> 2\. I'm wondering if the source for getting policies for a gateway should be in the `gateways` module. wdyt?

Fair enough. I’ve moved them in https://github.com/kumahq/kuma-gui/pull/1384/commits/57d8d4e0a1ccfb7dea95256d336bf988ea83fd20 and renamed the source path.

Reply:
Yeah I think keeping the prefixes the same as the module names (or at least related in the case of `gateways` vs `gateway`) and keeping them in their respective modules means you are far less likely to overwrite other things unless you actually want to.

Reply:
> Partially, yes, but the object returned isn’t just policies. It’s a MeshGatewayDataplane with gateway, listeners, and policies properties. I’m still fine with changing the data source paths, but it’s definitely an odd API landscape we’re dealing with here.

Ah I see! Glad I asked, hmm ok feels like this is maybe like the `routing` information rather than the information for the proxy itself.

Ok I think this is one to circle back on then, as long as we distinguished between the fact that this is for a `gateway` and not a `dataplane` (aka sidecar) then I think thats fine for now.

Reply:
Just to be clear I think theres more to be done here but I'm not totally sure what and I think it needs looking at with outside of a single GH PR, so feel free to resolve this.

Reply:
I agree, but personally I think that would be much easier to on our end if there was a dedicated Gateway resource that we could query (and this might be added at some point, see https://github.com/kumahq/kuma/issues/4358).